### PR TITLE
Set model in the eval mode for pruning

### PR DESCRIPTION
**Pull request by eldarkurtic**
Fixes https://github.com/neuralmagic/sparseml/issues/756

### CODE REVIEW

File: src/sparseml/pytorch/sparsification/pruning/modifier_pruning_mfac.py [hunk 1]
```diff
@@ -333,9 +333,13 @@ def _collect_grad_samples(
         )
 
         _LOGGER.debug("Starting to collect {num_grads} grads with GradSampler")
+        _LOGGER.info("Setting the model in the eval mode")
```
Comment (KSGulin) [inline src/sparseml/pytorch/sparsification/pruning/modifier_pruning_mfac.py:336]: Status updates should be logged in debug
```suggestion
        _LOGGER.debug("Setting the model in the eval mode")
```

File: src/sparseml/pytorch/sparsification/pruning/modifier_pruning_mfac.py [hunk 2]
```diff
@@ -333,9 +333,13 @@ def _collect_grad_samples(
...
+        module.eval()
         for _ in grad_sampler.iter_module_backwards(module, num_grads):
             self._module_masks.pre_optim_step_update()
         _LOGGER.debug("GradSampler grad collection complete")
+        _LOGGER.info("Setting the model back to the train mode")
```
Comment (KSGulin) [inline src/sparseml/pytorch/sparsification/pruning/modifier_pruning_mfac.py:341]: Status updates should be logged in debug
```suggestion
        _LOGGER.debug("Setting the model back to the train mode")
```

**Review by KSGulin — COMMENTED** [review]
The woodfisher computations should've definitely been done in eval mode. Thanks for catching this @eldarkurtic. Left two comments on the logging, otherwise LGTM

File: src/sparseml/pytorch/sparsification/pruning/modifier_pruning_mfac.py (hunk 1 above)
Comment (eldarkurtic) [reply src/sparseml/pytorch/sparsification/pruning/modifier_pruning_mfac.py]: Done

**Review by bfineran — COMMENTED** [review]
thanks for this fix @eldarkurtic, definitely should be in there

quick suggestion to check the modules' mode before `is_training = module.training` and only set back to train if `is_training`

**Review by bfineran — APPROVED** [review]

**Review by KSGulin — APPROVED** [review]

**Pull request merged**